<a href="https://colab.research.google.com/github/niteshchobdar38/EDA-project/blob/main/Nitesh_Play_Store_App_Review_Analysis_EDA_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Project Name**    -**Play store App Review Analysis** 



##### **Project Type**    - EDA
##### **Contribution**    - Individual


 **Project Summary** -The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market.
Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.
Explore and analyze the data to discover key factors responsible for app engagement and success. Google Play Store apps and reviews Mobile apps are everywhere. They are easy to create and can be lucrative. Because of these two factors, more and more apps are being developed. In this notebook, we will do a comprehensive analysis of the Android app market by comparing over ten thousand apps in Google Play across different categories. We'll look for insights in the data to devise strategies to drive growth and retention.

Let's take a look at the data, which consists of two Files:

playstore data.csv:  contains all the details of the applications on Google Play. There are 13 features that describe a given app.
user_reviews.csv:  contains 100 reviews for each app, most helpful first. The text in each review has been pre-processed and attributed with three new features: Sentiment (Positive, Negative or Neutral), Sentiment Polarity and Sentiment Subjectivity.


 **GitHub Link** -https://github.com/niteshchobdar38/EDA-project/blob/37660e82fa489921668a59dd2e9b00d0118bc5f7/Nitesh_Play_Store_App_Review_Analysis_EDA_Project.ipynb

# **Problem Statement**
**Buissness** **Problem** **Overview**

Play Store App Review Analysis is very important for any Buissness as it recognizes which app is highly demanded and useful.
- Software application is vital because  specific
software is required in almost every industry, in every
business ,and for each function. It become more important
as time goes on. Mobile app distribution platform such as
Google play store gets flooded with millions of new
applications uploaded by developers everyday. So in this
project, we aim on analyzing Google play store that provides
a particular app description and data such as reviews,
ratings, price and number of downloads. The objective of
this is to analyze the desire of the customer through the
reviews provided in the feedback section and apps trend in
the market to help the organization & developers. To this
end, we provide an idea about app that managed to get
maximum and minimum number of downloads and
predicting the category of apps that is most likely to be
downloaded in the coming years. Moreover, doing
sentimental analysis on the apps that generated most
positive and negative sentiments, sustainability of app in
market on basis of previous data and current market
situation. Furthermore, also analyzing the apps that has
maximum downloads have they managed to get average
rating so that concluding the co-relation between number of
downloads and ratings received


# **Business** **Objective**

The objective of this is to analyze the desire of the customer through the reviews provided in the feedback section and apps trend in the market to help the organization & developers.

# ***Let's Begin !***

### Import Libraries

In [1]:
# Import Libraries
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import numpy as np # linear algebra
import matplotlib.pyplot as plt
import seaborn as sns  # visualization tool
from datetime import datetime
%matplotlib inline

### Dataset Loading

In [2]:
# Load Dataset
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
#loading csv file 
ps_df = pd.read_csv("/content/drive/MyDrive/Play Store Data.csv") 
ur_df = pd.read_csv("/content/drive/MyDrive/User Reviews.csv")

### Dataset First View

In [5]:
# Dataset First Look
ps_df.head(1)

App        Category  Rating  \
0  Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   

  Reviews Size Installs  Type Price Content Rating        Genres  \
0     159  19M  10,000+  Free     0       Everyone  Art & Design   

      Last Updated Current Ver   Android Ver  
0  January 7, 2018       1.0.0  4.0.3 and up

### Dataset Rows & Columns count

In [6]:
# Dataset Rows & Columns count
ps_df.shape , ur_df.shape

((10841, 13), (64295, 5))

### Dataset Information

In [7]:
# Dataset Info
ps_df.info  , ur_df.info

(<bound method DataFrame.info of                                                      App             Category  \
 0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
 1                                    Coloring book moana       ART_AND_DESIGN   
 2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
 3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
 4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
 ...                                                  ...                  ...   
 10836                                   Sya9a Maroc - FR               FAMILY   
 10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
 10838                             Parkinson Exercices FR              MEDICAL   
 10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
 10840      iHoroscope - 2018 Daily Horoscope & Astrology         

#### Duplicate Values

In [ ]:
# Dataset Duplicate Value Count
len(ps_df[ps_df.duplicated()])

483

#### Missing Values/Null Values

In [ ]:
# Missing Values/Null Values Count
# This user define function will give the type,count of null and non null values as well as null ratio
def playstoreinfo():
  temp=pd.DataFrame(index=pd_df.columns)
  temp["datatype"]=pd_df.dtypes
  temp["not null values"]=pd_df.count()
  temp["null value"]=pd_df.isnull().sum()
  temp["% of the null value"]=pd_df.isnull().mean()
  temp["unique count"]=pd_df.nunique()
  return temp
playstoreinfo()


datatype  not null values  null value  % of the null value  \
App              object            10841           0             0.000000   
Category         object            10841           0             0.000000   
Rating          float64             9367        1474             0.135965   
Reviews          object            10841           0             0.000000   
Size             object            10841           0             0.000000   
Installs         object            10841           0             0.000000   
Type             object            10840           1             0.000092   
Price            object            10841           0             0.000000   
Content Rating   object            10840           1             0.000092   
Genres           object            10841           0             0.000000   
Last Updated     object            10841           0             0.000000   
Current Ver      object            10833           8             0.000738   
Android Ver      object            10838           3             0.000277   

                unique count  
App                     9660  
Category                  34  
Rating                    40  
Reviews                 6002  
Size                     462  
Installs                  22  
Type                       3  
Price                     93  
Content Rating             6  
Genres                   120  
Last Updated            1378  
Current Ver             2832  
Android Ver               33

Rating has 1474 null values which contributes 13.60% of the data. Type has 1 null value which contributes 0.01% of the data. Content_Rating has 1 null value which contributes 0.01% of the data. Current_Ver has 8 null values which contributes 0.07% of the data. Android_Ver has 3 null values which contributes 0.03% of the data.

In [ ]:
# Visualizing the missing values

### What did you know about your dataset?

Answer Here

## ***2. Understanding Your Variables***

In [ ]:
# Dataset Columns

In [ ]:
# Dataset Describe

### Variables Description 

Answer Here

### Check Unique Values for each variable.

In [ ]:
# Check Unique Values for each variable.

## 3. ***Data Wrangling***

### Data Wrangling Code

In [ ]:
# Write your code to make your dataset analysis ready.

### What all manipulations have you done and insights you found?

Answer Here.

## ***4. Data Vizualization, Storytelling & Experimenting with charts : Understand the relationships between variables***

#### Chart - 1

In [ ]:
# Chart - 1 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact? 
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 2

In [ ]:
# Chart - 2 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact? 
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 3

In [ ]:
# Chart - 3 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact? 
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 4

In [ ]:
# Chart - 4 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact? 
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 5

In [ ]:
# Chart - 5 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact? 
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 6

In [ ]:
# Chart - 6 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact? 
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 7

In [ ]:
# Chart - 7 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact? 
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 8

In [ ]:
# Chart - 8 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact? 
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 9

In [ ]:
# Chart - 9 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact? 
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 10

In [ ]:
# Chart - 10 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact? 
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 11

In [ ]:
# Chart - 11 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact? 
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 12

In [ ]:
# Chart - 12 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact? 
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 13

In [ ]:
# Chart - 13 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact? 
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 14 - Correlation Heatmap

In [ ]:
# Correlation Heatmap visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

#### Chart - 15 - Pair Plot 

In [ ]:
# Pair Plot visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

## **5. Solution to Business Objective**

#### What do you suggest the client to achieve Business Objective ? 
Explain Briefly.

Answer Here.

# **Conclusion**

Write the conclusion here.

### ***Hurrah! You have successfully completed your EDA Capstone Project !!!***